sale_commission/migrations/10.0.2.4.0/pre-populate_company.py

```diff
@@ -0,0 +1,23 @@
+#  -*- coding: utf-8 -*-
+#  Copyright 2019 Simone Rubino - Agile Business Group
+#  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+def migrate(cr, installed_version):
+    """Populate company field for settlements and their lines"""
+    # Populate company_id for sale.commission.settlement,
+    # get the company from the settlement's agent's company
+    cr.execute(
+        'UPDATE sale_commission_settlement '
+        'SET company_id = res_partner.company_id'
+        'FROM res_partner'
+        'WHERE sale_commission_settlement.agent = res_partner.id')
+
+    # Populate company_id for sale.commission.settlement.line,
+    # get the company from the settlement
+    cr.execute(
+        'UPDATE sale_commission_settlement_line '
+        'SET company_id = sale_commission_settlement.company_id'
+        'FROM sale_commission_settlement'
+        'WHERE sale_commission_settlement_line.settlement '
+        '   = sale_commission_settlement.id')
```

Comment:
Can this be named `pre-populate_company.py`? Did you test it?

Reply:
Yes, actually it can be named `pre-<whatever>` for being executed.

Reply:
AFAIK the only requirement for ensuring the script migrations will be executed is that the python file names must start by `pre` or `post`.

Reply:
Actually it's `pre-` / `post-` (with dash).

Reply:
> Actually it's `pre-` / `post-` (with dash).

Correct, from https://github.com/odoo/odoo/blob/7746621b92c27f6250c0895b0c54b501097e05bb/odoo/modules/migration.py#L28

Reply:
I know it because I was the one that needed to correct that in Odoo documentation: https://github.com/odoo/odoo/commit/c0e35f84e23ebda8ea6033711ded5638ef309da2 :wink:

Reply:
> Can this be named `pre-populate_company.py`? Did you test it?

I tested the queries in postgresql, did not test the migration execution.
Now I have tried this and it says
```
ProgrammingError: column "company_id" of relation "sale_commission_settlement_line" does not exist
LINE 1: UPDATE sale_commission_settlement_line SET company_id = sale...
```

Maybe the script should be a post-<stuff>?

Reply:
Yes, as the column is populated on the update itself (after pre and before post).

Reply:
Changed to post- and error was still there, then removed the query on the lines because I hadn't noticed that it is a related field (non-stored by default)

Reply:
@SimoRubi So is it working as expected according to your tests now?

Reply:
Yes in my local environment it is working as expected it is @eLBati